ui/components/TopMenu/TopMenuProfileButton.tsx

```diff
@@ -1,31 +1,78 @@
 import { selectCurrentAccountTotal } from "@tallyho/tally-background/redux-slices/selectors"
-import React, { ReactElement } from "react"
+import { setSnackbarMessage } from "@tallyho/tally-background/redux-slices/ui"
+import React, { ReactElement, useState, useRef } from "react"
+import { useDispatch } from "react-redux"
 import { useBackgroundSelector } from "../../hooks"
 import SharedCurrentAccountInformation from "../Shared/SharedCurrentAccountInformation"
+import TopMenuProfileTooltip from "./TopMenuProfileTooltip"
+
+const TOOLTIP_DELAY = 500
 
 export default function TopMenuProfileButton(props: {
   onClick?: () => void
 }): ReactElement {
-  const { shortenedAddress, name, avatarURL } =
+  const dispatch = useDispatch()
+  const { shortenedAddress, name, avatarURL, address } =
     useBackgroundSelector(selectCurrentAccountTotal) ?? {}
 
+  const [shouldDisplayTooltip, setShouldDisplayTooltip] = useState(false)
+  const timerRef = useRef<number | undefined>(undefined)
+
   const { onClick } = props
 
+  const showTooltip = () => {
+    timerRef.current = window.setTimeout(
+      () => setShouldDisplayTooltip(true),
+      TOOLTIP_DELAY
+    )
+  }
+  const hideTooltip = () => {
+    clearTimeout(timerRef.current ?? 0)
```

Comment:
Better be more explicit:
```suggestion
    if (typeof timerRef.current !== undefined) clearTimeout(timerRef.current)
    timerRef.current = undefined
```

Also, using `null` instead of `undefined` as default avoids the need of `typeof ...` (which I think is not necessary anyway - unless we want to replace all the occurrences of `undefined` with `void 0`)